docs/rules/prefer-nullish-coalescing.md

```diff
@@ -0,0 +1,27 @@
+# Prefer the nullish coalescing operator(`??`) over the logical OR operator(`||`)
+
```

Comment:
There are some new requirements for rule docs, be sure to rebase on the latest `main` to pick them up.